nixos/modules/services/games/minecraft-server.nix

```diff
@@ -61,13 +143,26 @@ in
       serviceConfig.User    = "minecraft";
       script = ''
         cd ${cfg.dataDir}
+
+        ln -sf ${eulaFile} eula.txt
+        ${if cfg.whitelist != null then "ln -sf ${whitelistFile} whitelist.json" else ""}
+
+        # create backup of original server.properties
+        if [ -f "server.properties" ]; then
+            cp -n server.properties server.properties.old
+        fi
+        cp ${serverPropertiesFile} server.properties
+        # server.properties must have write permissions or else it crashes
+        chmod 700 server.properties
```

Comment:
This would overwrite group permissions if the user has set them. Just `chmod +w server.properties` would be better.